### ISSUE TITLE
[Suggestion] Add embedded stackblitz examples on documentation page

### ISSUE THREAD

**Issue by vneogi199** [nartc/ngxtension-platform#148]
Creating interactive examples will help users to understand the functionality of each utility.

I got this idea from [VueUse](https://vueuse.org/core/useActiveElement/). I'm not sure if its possible to embed complete Angular application within Astro. But we can embed Stackblitz examples to help users to play with each utility in the browser.

<img width="865" alt="image" src="https://github.com/nartc/ngxtension-platform/assets/20491952/392656d5-9bd3-41fd-b553-aaff5a8037f0">

**Comment by nartc** [comment]
As of this moment, no. There's no straightforward way to render Angular components in Markdown yet (Brandon might come up with something though, I'll check with him).

Embedded Stackblitz sounds good to me but managing the examples can be tricky. Do you have any suggestions on managing Stackblitz examples?

**Comment by vneogi199** [comment]
I could render Angular component in plain Astro app by using [Analog's Astro plugin](https://analogjs.org/docs/packages/astro-angular/overview#using-components-in-mdx-pages). But I cant get the plugin working with Starlight (it breaks loading the homepage after adding the plugin, even without loading any Angular component).

```
TypeError: Cannot read properties of undefined (reading 'hidden')
    at /Users/vinit/Desktop/Angular/ngxtension-platform/docs/node_modules/.pnpm/@astrojs+starlight@0.9.1_astro@3.0.13/node_modules/@astrojs/starlight/utils/navigation.ts:151:44
    at Array.filter (<anonymous>)
    at treeify (/Users/vinit/Desktop/Angular/ngxtension-platform/docs/node_modules/.pnpm/@astrojs+starlight@0.9.1_astro@3.0.13/node_modules/@astrojs/starlight/utils/navigation.ts:151:4)
    at groupFromAutogenerateConfig (/Users/vinit/Desktop/Angular/ngxtension-platform/docs/node_modules/.pnpm/@astrojs+starlight@0.9.1_astro@3.0.13/node_modules/@astrojs/starlight/utils/navigation.ts:92:15)
```
I feel like I'm missing some configuration somewhere.
I am in favor of rendering Angular directly instead of Stackblitz as it will help us prevent updating the Starblitz examples everytime we make a new version.

**Comment by michael-small** [comment]
https://blog.stackblitz.com/posts/announcing-tutorialkit/

You know the "Learn Angular" format on the new docs? Info on the left and live code on the right? That's what this is like, and it runs a real app directly in Astro.

I got this working with an Angular example. I can deploy it if anyone is interested, otherwise just run it locally: https://github.com/michael-small/angular-TutorialKit-hello-world

![image](https://github.com/user-attachments/assets/0d2d3e61-0d6c-460c-9b39-5b90c2489fef)

Basically, there could be a project for the docs, and then for a given util it is embedded in a component. Then the doc site's header points at a copy of that component which is inside of the given doc page. That doc page runs the app with that instance of the component, embedded in a preview, and editable.

My example was just my hello world, but that is a copy of the `app.component.html` inside of "Basics > Introduction > Welcome to TutorialKit" section

**Comment by michael-small** [comment]
https://ng-tutkit-ngxtension-ms.firebaseapp.com/ngxtension/injectables/injectqueryparams/
https://github.com/michael-small/angular-TutorialKit-hello-world

Disclaimer: I can change the name, but that's what I called my firebase project. Not a project owner lol.

I made and deployed examples for `injectParams` and `injectQueryParams`
- Documentation copy is verbatim from the ngxtension docs.
- Example code that is run on the right is somewhat similar but diverges from the doc's code. I just wanted to get something going to show.

I am still figuring out the basics of TutorialKit integration of Angular, as well as hosting with Firebase in general, so this is a bit rough. Also I am bad at routing in general so I made some weird choices as stuff only happened while it was deployed. But it works.

Thoughts?

**Comment by nartc** [comment]
@michael-small nice work! 

The internal app is for utilities that make more sense to be tested in an e2e environment like Resize. 

I'd love this for our docs but we need to discuss about the DX part of this like how we going to support the current utilities and make adding new utilities easy.

**Comment by msmallest** [comment]
Thank you

> The internal app is for utilities that make more sense to be tested in an e2e environment like Resize.

Yeah, I liked the internal one for testing out my feature as I made it. Makes sense it would be better for e2e.

> I'd love this for our docs but we need to discuss about the DX part of this like how we going to support the current utilities and make adding new utilities easy.

Yeah, I imagined this would be a big commitment to consider and implement and build into the future flows. 

My intent with this deployment other than this being a proof of concept for ngxtension if deemed a good/viable resource was getting used to TutorialKit for Angular and documenting what I use in work projects. Going down the line I intend to knock out `filterNil/signalSlice/navigationEnd/callApply/rxEffect` for my own reference and my team. So if nothing comes of this officially for ngxtension, I intend to keep up this firebase deploy for those functions.

**Comment by nartc** [comment]
@msmallest I honestly have no time to focus on this so if you'd like to spear head this, I'll support you in every way I can. Otherwise, this has to wait a bit longer.

**Comment by Dafnik** [comment]
@michael-small Do you still have the source code up somewhere? The GitHub link is sadly broken.

I've would have really liked to take a look at the setup you've used.

**Comment by michael-small** [comment]
@Dafnik I don't know why it was private, I thought it was public, sorry. Should be public now.

I haven't done much in the last couple weeks on it but there is some utils on there so far.
